packages/contracts-bedrock/test/invariants/interop/Setup.sol

```diff
@@ -0,0 +1,164 @@
+import { IOptimismPortal2 } from "interfaces/L1/IOptimismPortal2.sol";
+import { IOptimismPortalInterop } from "interfaces/L1/IOptimismPortalInterop.sol";
+import { ISharedLockbox } from "interfaces/L1/ISharedLockbox.sol";
+import { ISuperchainConfig } from "interfaces/L1/ISuperchainConfig.sol";
+import { ISuperchainWETH } from "interfaces/L2/ISuperchainWETH.sol";
+import { ISystemConfigInterop } from "interfaces/L1/ISystemConfigInterop.sol";
+
+// Interfaces 0.8.25
+import { ICrossL2Inbox } from "interfaces/L2/ICrossL2Inbox.sol";
+import { IL2ToL2CrossDomainMessenger } from "interfaces/L2/IL2ToL2CrossDomainMessenger.sol";
+import { ISuperchainERC20 } from "interfaces/L2/ISuperchainERC20.sol";
+import { ISuperchainTokenBridge } from "interfaces/L2/ISuperchainTokenBridge.sol";
+
+// Libraries and Constants
+import { Constants } from "src/libraries/Constants.sol";
+import { GameType } from "src/dispute/lib/Types.sol";
+import { IDisputeGameFactory } from "interfaces/dispute/IDisputeGameFactory.sol";
+import { ISystemConfig } from "interfaces/L1/ISystemConfig.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+
+contract Setup {
+    IStdCheats constant vm = IStdCheats(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+    IDeployer815 constant deployer815 = IDeployer815(0x4200000000000000000000000000000000000815);
+    IDeployer825 constant deployer825 = IDeployer825(0x4200000000000000000000000000000000000825);
+
+    bytes32 internal constant _IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+    bytes32 internal constant _ADMIN_SLOT = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+
+    // Solidity 0.8.15 Contracts
+    IETHLiquidity constant ethLiquidity = IETHLiquidity(Predeploys.ETH_LIQUIDITY);
+    IL1BlockInterop constant l1BlockInterop = IL1BlockInterop(Predeploys.L1_BLOCK_ATTRIBUTES);
+    ILiquidityMigrator immutable liquidityMigrator;
+    IOptimismPortal2 immutable optimismPortal2;
+    IOptimismPortalInterop immutable optimismPortalInterop;
+    ISharedLockbox immutable sharedLockbox;
+    ISuperchainConfig immutable superchainConfig;
+    ISuperchainWETH constant superWeth = ISuperchainWETH(payable(Predeploys.SUPERCHAIN_WETH));
+    ISystemConfigInterop immutable systemConfigInterop;
+
+    // Soldity 0.8.25 Contracts
+    ICrossL2Inbox constant inbox = ICrossL2Inbox(Predeploys.CROSS_L2_INBOX);
+    IL2ToL2CrossDomainMessenger constant messenger =
+        IL2ToL2CrossDomainMessenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
+    ISuperchainERC20 immutable medusaToken;
+    ISuperchainTokenBridge constant tokenBridge = ISuperchainTokenBridge(Predeploys.SUPERCHAIN_TOKEN_BRIDGE);
+
+    // Actors
+    address immutable guardian = vm.addr(uint256(keccak256("Guardian")));
+    address immutable upgrader = vm.addr(uint256(keccak256("Upgrader")));
+    address immutable admin = vm.addr(uint256(keccak256("Admin")));
```

Comment:
Isn't this the proxy admin?

We need to update the var name, but also to deploy the `ProxyAdmin` since it is a contract and not an EOA

Reply:
Does it change anything to our test if the Admin is a contract or a EOA?

Reply:
I mean, we are not testing the proxies and CHUGSPLASH and ERC1967 store the implementation in the same slot and both use delegatecall. Not sure why we would need to use ProxyAdmin contract since we are setting the implementation and admin with `vm.store`.